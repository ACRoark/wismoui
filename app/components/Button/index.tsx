import React from 'react';

import classnames from 'classnames';

export type ButtonHtmlType = 'button' | 'reset' | 'submit';
export type ButtonTypes = 'link' | 'primary' | 'secondary';

export interface IButtonProps {
  children?: React.ReactNode;
  className?: string;
  dataTrackSubAction?: string;
  disabled?: boolean;
  htmlType?: ButtonHtmlType;
  id?: string;
  onClick?: () => void;
  type?: ButtonTypes;
}

const getClasses = (className?: string, type?: ButtonTypes): string =>
  classnames(className, {
    'dte-wismo-button': true,
    'dte-wismo-button-link': type === 'link',
    'dte-wismo-button-primary': !type || type === 'primary',
    'dte-wismo-button-secondary': type === 'secondary',
  });

const Button: React.FC<IButtonProps> = (props: IButtonProps): React.ReactElement => {
  const { children, className, dataTrackSubAction, disabled, htmlType, id, onClick, type } = props;

  const classes: string = getClasses(className, type);

  return (
    // eslint-disable-next-line react/button-has-type
    <button
      className={classes}
      disabled={disabled}
      data-track="mimo"
      data-track-action="click"
      data-track-detail="order tracker"
      data-track-sub-action={dataTrackSubAction}
      id={id}
      onClick={onClick}
      type={htmlType}
    >
      {React.Children.toArray(children)}
    </button>
  );
};

export default Button;
