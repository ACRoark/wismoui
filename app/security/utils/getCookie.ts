const getCookie = (name: string): string =>
  document.cookie.split('; ').reduce((previousValue: string, currentValue: string): string => {
    const parts = currentValue.split('=');

    return parts[0] === name ? decodeURIComponent(parts[1]) : previousValue;
  }, '');

export default getCookie;
