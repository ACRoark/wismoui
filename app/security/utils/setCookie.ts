interface ICookieOptions {
  days: number;
  path: string;
}

const setCookie = (name: string, value: string, options: ICookieOptions): void => {
  const expires = new Date(Date.now() + options.days * 864e5).toUTCString();

  document.cookie = `${name}=${encodeURIComponent(value)}; expires=${expires}; path=${options.path}`;
};

export default setCookie;
