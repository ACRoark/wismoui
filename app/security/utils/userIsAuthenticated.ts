import useAuth from 'hooks/useAuth';

const userIsAuthenticated = (): boolean => {
  const { isAuthenticated } = useAuth();

  return isAuthenticated;
};

export default userIsAuthenticated;
