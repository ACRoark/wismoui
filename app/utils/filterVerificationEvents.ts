import IVerificationEvent from 'types/IVerificationEvent';
import filterVerificationEventStatusUpdatesByActive from './filterVerificationEventStatusUpdatesByActive';
import filterVerificationEventStatusUpdatesByVerificationCodes from './filterVerificationEventStatusUpdatesByVerificationCodes';

const filterVerificationEvents = (verificationEvents: IVerificationEvent[], verificationCodes: string[]): IVerificationEvent[] => {
  const filteredVerificationEvents: IVerificationEvent[] = [];
  verificationEvents.forEach((event: IVerificationEvent): void => {
    const activeVerificationEventStatusUpdates = filterVerificationEventStatusUpdatesByActive(event.verificationEventStatusUpdates);
    const verificationEventStatusUpdatesByVerificationCodes =
      filterVerificationEventStatusUpdatesByVerificationCodes(activeVerificationEventStatusUpdates, verificationCodes);

    if (verificationEventStatusUpdatesByVerificationCodes.length > 0) {
      const filteredEvent = {
        ...event,
        verificationEventStatusUpdates: verificationEventStatusUpdatesByVerificationCodes
      };

      filteredVerificationEvents.push(filteredEvent);
    }
  });

  return filteredVerificationEvents;
};

export default filterVerificationEvents;
