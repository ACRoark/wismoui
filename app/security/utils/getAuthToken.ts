import getCookie from './getCookie';

const getAuthToken = (): string => getCookie('token');

export default getAuthToken;
