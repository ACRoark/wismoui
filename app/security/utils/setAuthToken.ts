import setCookie from './setCookie';

const setAuthToken = (value?: string): void => {
  if (value) {
    setCookie('token', `Bearer ${value}`, { days: 1, path: '/' });
  }
};

export default setAuthToken;
