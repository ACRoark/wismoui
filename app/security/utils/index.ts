import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
