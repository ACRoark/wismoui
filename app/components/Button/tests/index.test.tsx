import { render } from 'enzyme';
import React from 'react';

import { createSnapshot } from 'testing/utils';

import Button, { ButtonTypes } from '..';

const expectButtonToMatchSnapshot = (type?: ButtonTypes, disabled?: boolean, dataTrackSubAction?: string): void => {

  const tree = createSnapshot(<Button disabled={disabled} type={type} dataTrackSubAction={dataTrackSubAction} />);

  expect(tree).toMatchSnapshot();
};

describe('<Button />', (): void => {

  it('should not log errors in console when rendered', (): void => {
      const spy = jest.spyOn(global.console, 'error');

      render(<Button type="primary" />);

      expect(spy).not.toHaveBeenCalled();
    });

  it('should render correctly when it is link', (): void => expectButtonToMatchSnapshot('link'));

  it('should render correctly when it is link and disabled', (): void => expectButtonToMatchSnapshot('link', true));

  it('should render correctly when it is primary', (): void => expectButtonToMatchSnapshot('primary'));

  it('should render correctly when it is primary and disabled', (): void => expectButtonToMatchSnapshot('primary', true));

  it('should render correctly when it is secondary', (): void => expectButtonToMatchSnapshot('secondary'));

  it('should render correctly when it is secondary and disabled', (): void => expectButtonToMatchSnapshot('secondary', true));

  it('should render correctly when the type is not specified', (): void => expectButtonToMatchSnapshot());

  it('should render correctly when the type is not specified and the button is disabled', (): void => expectButtonToMatchSnapshot(undefined, true));
});
