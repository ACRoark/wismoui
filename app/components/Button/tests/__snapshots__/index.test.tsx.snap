// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly when it is link 1`] = `
<button
  className="dte-wismo-button dte-wismo-button-link"
  data-track="mimo"
  data-track-action="click"
  data-track-detail="order tracker"
/>
`;

exports[`<Button /> should render correctly when it is link and disabled 1`] = `
<button
  className="dte-wismo-button dte-wismo-button-link"
  data-track="mimo"
  data-track-action="click"
  data-track-detail="order tracker"
  disabled={true}
/>
`;

exports[`<Button /> should render correctly when it is primary 1`] = `
<button
  className="dte-wismo-button dte-wismo-button-primary"
  data-track="mimo"
  data-track-action="click"
  data-track-detail="order tracker"
/>
`;

exports[`<Button /> should render correctly when it is primary and disabled 1`] = `
<button
  className="dte-wismo-button dte-wismo-button-primary"
  data-track="mimo"
  data-track-action="click"
  data-track-detail="order tracker"
  disabled={true}
/>
`;

exports[`<Button /> should render correctly when it is secondary 1`] = `
<button
  className="dte-wismo-button dte-wismo-button-secondary"
  data-track="mimo"
  data-track-action="click"
  data-track-detail="order tracker"
/>
`;

exports[`<Button /> should render correctly when it is secondary and disabled 1`] = `
<button
  className="dte-wismo-button dte-wismo-button-secondary"
  data-track="mimo"
  data-track-action="click"
  data-track-detail="order tracker"
  disabled={true}
/>
`;

exports[`<Button /> should render correctly when the type is not specified 1`] = `
<button
  className="dte-wismo-button dte-wismo-button-primary"
  data-track="mimo"
  data-track-action="click"
  data-track-detail="order tracker"
/>
`;

exports[`<Button /> should render correctly when the type is not specified and the button is disabled 1`] = `
<button
  className="dte-wismo-button dte-wismo-button-primary"
  data-track="mimo"
  data-track-action="click"
  data-track-detail="order tracker"
  disabled={true}
/>
`;
